src/types/postWithAllDate.ts

```diff
@@ -0,0 +1,12 @@
+import { Comment } from './comment';
+import { Post } from './post';
+import { User } from './user';
+
+export interface PostWithAllDate extends Post {
+  userId: number,
+  id: number,
+  title: string,
+  body: string,
```

Comment:
your Post already contains all this fields, you don't need it here, they will be inherited from PostWithAllDate